frontend/src/lib/internalMetrics.ts

```diff
@@ -9,14 +9,20 @@ interface InternalMetricsPayload {
 }
 
 interface TimeToSeeDataPayload {
-    query_id: string
-    status: 'failure' | 'success'
+    type: 'insight_load' | 'dashboard_load'
+    context: 'insight' | 'dashboard'
     time_to_see_data_ms: number
-    cached: boolean
-    current_url: string
+    dashboard_query_id?: string
+    query_id?: string
+    status?: 'failure' | 'success'
     api_response_bytes?: number
     api_url?: string
-    insight: string
+    insight?: string
+    action?: string
+    insights_fetched: number
+    insights_fetched_cached: number
+    min_last_refresh?: string | null
```

Comment:
this makes me wonder about storing an array of the refresh times, but since we're not worried about getting this schema right first time, I'm probably reaching too far into the future

Reply:
Arrays are annoying to analyze in sql, especially with nulls in them.

First iteration actually tried to store the array, but clickhouse-kafka didn't handle it gracefully either.